lua/lvim/lsp/templates.lua

```diff
@@ -17,13 +17,22 @@ end
 
 local skipped_filetypes = lvim.lsp.automatic_configuration.skipped_filetypes
 local skipped_servers = lvim.lsp.automatic_configuration.skipped_servers
+local ensure_installed_servers = lvim.lsp.installer.setup.ensure_installed
 
 ---Generates an ftplugin file based on the server_name in the selected directory
 ---@param server_name string name of a valid language server, e.g. pyright, gopls, tsserver, etc.
 ---@param dir string the full path to the desired directory
 function M.generate_ftplugin(server_name, dir)
   if vim.tbl_contains(skipped_servers, server_name) then
-    return
+    if not vim.tbl_contains(ensure_installed_servers, server_name) then
+      return
+    else
+      -- ensure_installed should take priority over skipped_servers
+      -- remove server from skipped_servers list if it is ensured install
+      lvim.lsp.automatic_configuration.skipped_servers = vim.tbl_filter(function(s)
+        return s ~= server_name
+      end, lvim.lsp.automatic_configuration.skipped_servers)
+    end
```

Comment:
I'm not sure what the best way to handle this is, I think this might get confusing to debug if we start modifying this list, keep in mind that this template generator doesn't get called all that often.

However I see the benefit, since `LvimInfo` also relies on this list.

If you want to go with this approach of modifying the skip list, then you need to do it in this `config:load()`, instead
https://github.com/LunarVim/LunarVim/blob/958030e728d3a853cc85051edf0326593c24148f/lua/lvim/config/init.lua#L104-L106

or create a new `should_skip()` function in the template generator, that checks all those hierarchy rules without modifying any of the original lists.

cc: @abzcoding, @rebuilt

Reply:
Then i should go with the second approach. Which do you think is better?

```lua
function M.generate_ftplugin(server_name, dir)
  if vim.tbl_contains(skipped_servers, server_name) 
      and not vim.tbl_contains(ensure_installed_servers, server_name) then
    return
  end
-- the rest of templates generator
```

or 

```lua
local function should_skip(server_name)
  return vim.tbl_contains(skipped_servers, server_name) 
      and not vim.tbl_contains(ensure_installed_servers, server_name)
end

function M.generate_ftplugin(server_name, dir)
  if should_skip(server_name) then
    return
  end
-- the rest of templates generator
```

Reply:
I'd go with with the second version, because there's also the case of skipped filetypes 😅 

https://github.com/LunarVim/LunarVim/blob/283e38b733c29514ee602ae3c251a2d9f4a9c053/lua/lvim/lsp/templates.lua#L38-L45

it's more for readability than anything, and would make it easier to add some debug log entries

Reply:
@cpea2506, rebasing on rolling should now fix the CI issues, sorry about that!

Reply:
Ah! Okay. Im crying with that thing! 😭